app/src/main/java/com/example/vshcheglov/webshop/presentation/login/LoginActivity.kt

```diff
@@ -28,29 +27,28 @@ class LoginActivity : NucleusAppCompatActivity<LoginPresenter>(), LoginPresenter
         buttonLogin.setOnClickListener {
             emailTextInput.error = ""
             passwordTextInput.error = ""
-            presenter.logOnUser(loginEmail.text.toString(), loginPassword.text.toString())
+            presenter.logInUser(loginEmail.text.toString(),
+                loginPassword.text.toString(), isNetworkAvailable())
         }
 
         buttonRegister.setOnClickListener {
             presenter.registerUser()
         }
-        initShowPasswordButton()
-    }
-
-    private fun initShowPasswordButton() {
         loginShowPasswordButton.setOnTouchListener { _, event ->
-            when(event.action) {
+            when (event.action) {
                 MotionEvent.ACTION_DOWN -> loginPassword.inputType = InputType.TYPE_CLASS_TEXT
-                MotionEvent.ACTION_UP -> loginPassword.inputType = InputType.TYPE_CLASS_TEXT or InputType.TYPE_TEXT_VARIATION_PASSWORD
+                MotionEvent.ACTION_UP -> loginPassword.inputType =
+                    InputType.TYPE_CLASS_TEXT or InputType.TYPE_TEXT_VARIATION_PASSWORD
             }
             true
         }
     }
 
     override fun startMainActivity() {
-        val intent = Intent(this, MainActivity::class.java)
-        intent.flags = Intent.FLAG_ACTIVITY_CLEAR_TASK or Intent.FLAG_ACTIVITY_NEW_TASK
-        startActivity(intent)
+        Intent(this, MainActivity::class.java).apply {
```

Comment:
startActivity(Intent(this, MainActivity::class.java).apply {
       flags = Intent.FLAG_ACTIVITY_CLEAR_TASK or Intent.FLAG_ACTIVITY_NEW_TASK 
})